core/src/wsv.rs

```diff
@@ -325,14 +339,45 @@ impl<W: WorldTrait> WorldStateView<W> {
             .map(|block_entry| block_entry.value().clone())
     }
 
-    /// Get an immutable view of the `World`.
-    pub fn world(&self) -> &W {
-        &self.world
-    }
+    /// Get `World` and pass it to closure to modify it
+    ///
+    /// Produces events after `f` finishes. Events are emitted from the lowest to the highest
+    ///
+    /// # Errors
+    /// Fails if `f` fails
+    pub fn modify_world(
+        &self,
+        f: impl FnOnce(&World) -> Result<WorldEvent, Error>,
+    ) -> Result<(), Error> {
+        let mut events: SmallVec<[DataEvent; 3]> = SmallVec(smallvec::smallvec![]);
+        let event = f(&self.world)?;
+
+        match &event {
+            WorldEvent::Domain(domain_event) => {
+                match domain_event {
+                    DomainEvent::Account(account_event) => {
+                        match account_event {
+                            AccountEvent::Asset(asset_event) => {
+                                events.push(DataEvent::Asset(asset_event.clone()))
+                            }
+                            AccountEvent::OtherAccountChange(_) => (),
```

Comment:
In this particular case, I would leave the enum variants specific, annotate them with `#[non_exhaustive]`, and leave this as a wildcard in the match statements. 

Having `Other` or `Miscellaneous` is a technical debt with a high interest rate. Instead of adding the correct specific variant, people tend to "stringly-type" the program and shove everything into other and miscellaneous.